framework/src/test/java/org/tron/core/db/AccountAssetStoreTest.java

```diff
@@ -0,0 +1,139 @@
+import org.tron.core.config.args.Args;
+import org.tron.core.store.AccountAssetStore;
+import org.tron.core.store.AccountStore;
+import org.tron.protos.Protocol;
+import org.tron.protos.contract.AssetIssueContractOuterClass;
+
+public class AccountAssetStoreTest extends BaseTest {
+
+  private static final byte[] ASSET_KEY = "20000".getBytes();
+  private static AccountCapsule ownerCapsule;
+
+  private static String OWNER_ADDRESS = Wallet.getAddressPreFixString()
+          + "abd4b9367799eaa3197fecb144eb71de1e049abc";
+  private static final long TOTAL_SUPPLY = 1000_000_000L;
+  private static final int TRX_NUM = 10;
+  private static final int NUM = 1;
+  private static final long START_TIME = 1;
+  private static final long END_TIME = 2;
+  private static final int VOTE_SCORE = 2;
+  private static final String DESCRIPTION = "TRX";
+  private static final String URL = "https://tron.network";
+
+  @Resource
+  private AccountAssetStore accountAssetStore;
+
+  @Resource
+  private AccountStore accountStore;
+
+  static {
+    dbPath = "db_AccountAssetStore_test";
+    Args.setParam(
+            new String[]{
+                "--output-directory", dbPath,
+            },
+            Constant.TEST_CONF
+    );
+  }
+
+  @Before
+  public void init() {
+    accountAssetStore.put(ASSET_KEY, Longs.toByteArray(200L));
+
+    ownerCapsule =
+            new AccountCapsule(
+                    ByteString.copyFrom(ByteArray.fromHexString(OWNER_ADDRESS)),
+                    ByteString.copyFromUtf8("owner"),
+                    Protocol.AccountType.AssetIssue);
+  }
+
+
+  private long createAsset(String tokenName) {
+    long id = chainBaseManager.getDynamicPropertiesStore().getTokenIdNum() + 1;
+    chainBaseManager.getDynamicPropertiesStore().saveTokenIdNum(id);
+    AssetIssueContractOuterClass.AssetIssueContract assetIssueContract =
+            AssetIssueContractOuterClass.AssetIssueContract.newBuilder()
+                    .setOwnerAddress(ByteString.copyFrom(ByteArray.fromHexString(OWNER_ADDRESS)))
+                    .setName(ByteString.copyFrom(ByteArray.fromString(tokenName)))
+                    .setId(Long.toString(id))
+                    .setTotalSupply(TOTAL_SUPPLY)
+                    .setTrxNum(TRX_NUM)
+                    .setNum(NUM)
+                    .setStartTime(START_TIME)
+                    .setEndTime(END_TIME)
+                    .setVoteScore(VOTE_SCORE)
+                    .setDescription(ByteString.copyFrom(ByteArray.fromString(DESCRIPTION)))
+                    .setUrl(ByteString.copyFrom(ByteArray.fromString(URL)))
+                    .build();
+    AssetIssueCapsule assetIssueCapsule = new AssetIssueCapsule(assetIssueContract);
+    chainBaseManager.getAssetIssueV2Store()
+            .put(assetIssueCapsule.createDbV2Key(), assetIssueCapsule);
+    try {
+      ownerCapsule.addAssetV2(ByteArray.fromString(String.valueOf(id)), TOTAL_SUPPLY);
+    } catch (Exception e) {
+      e.printStackTrace();
```

Comment:
Should this be thrown?

Reply:
It is possible not to throw this exception

Reply:
Shall it be throw out but not be caught ?